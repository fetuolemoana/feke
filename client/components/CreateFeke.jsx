import React from 'react'
import { connect } from 'react-redux'
import { createFeke } from '../actions/createFeke'
import { Redirect } from 'react-router'

class CreateFeke extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      category_id: '1',
      name: '',
      village: 'Avatele',
      image: 'feke-blue.png',
      status: 'Level 1'
    }

    this.handleSubmit = this.handleSubmit.bind(this)
    this.handleOnclick = this.handleOnclick.bind(this)
    this.handleDropdown = this.handleDropdown.bind(this)
    this.handleChange = this.handleChange.bind(this)
  }

  handleOnclick(e) {
    this.setState({
      image: e.target.name
    })
  }

  handleDropdown(e) {
    this.setState({
      village: e.target.value
    })
  }

  handleChange(e) {
    this.setState({
      name: e.target.value
    })
  }

  handleSubmit = () => {
    let feke = this.state
    this.props.dispatch(createFeke(feke))
  }

  render() {
    if (this.props.isUpdated) {
      return <Redirect to= '/category'/>
    }

    return (
      <div className="createfeke-page">
        <form onSubmit={this.handleSubmit}>
        <div className='createfeke-container'>
          <img
            className="createfeke-image"
            src="images/feke-blue.png"
            name="feke-blue.png"
            onClick={this.handleOnclick}
          />
          <img
            className="createfeke-image"
            src="images/feke-green.png"
            name="feke-green.png"
            onClick={this.handleOnclick}
          />
          <img
            className="createfeke-image"
            src="images/feke-purple.png"
            name="feke-purple.png"
            onClick={this.handleOnclick}
          />
          <img
            className="createfeke-image"
            src="images/feke-red.png"
            name="feke-red.png"
            onClick={this.handleOnclick}
          />
  </div>
  <div className="createfeke-container">
          <label htmlFor="name">Name </label>
          <input
            type="text"
            name="name"
            value={this.state.name}
            onChange={this.handleChange}
          />

          <label htmlFor="village">Village </label>
          <select value={this.state.village} onChange={this.handleDropdown}>
            <option village="alofi">Alofi</option>
            <option village="avatele">Avatele</option>
            <option village="hakupu">Hakupu</option>
            <option village="hikutavake">Hikutavake</option>
            <option village="lakepa">Lakepa</option>
            <option village="liku">Liku</option>
            <option village="makefu">Makefu</option>
            <option village="mutalau">Mutalau</option>
            <option village="namukulu">Namukulu</option>
            <option village="tamakautoga">Tamakautoga</option>
            <option village="toi">Toi</option>
            <option village="tuapa">Tuapa</option>
            <option village="vaiea">Vaiea</option>
          </select>
           <input type="submit" value="Create feke" />
           </div>
        </form>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  isUpdated: state.isUpdated
})

export default connect(mapStateToProps)(CreateFeke)