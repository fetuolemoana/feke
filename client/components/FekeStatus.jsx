import React from 'react'
import {connect} from 'react-redux'
import {getFekeStatus} from '../actions/getFekeStatus'

class FekeStatus extends React.Component {
  componentDidMount () {
    this.props.dispatch(getFekeStatus())
  }
  render () {
    return (
      <div className = 'fekeStatus'>
        <h1>This is the Feke Status Component</h1>
        <div>{this.props.feke.map((feke) => {
          return (

            <div key = {feke.name}>
              <img src = {`/images/${feke.image}`} />
              <h1>{feke.name}</h1>
              <h1>{feke.village}</h1>
            </div>
          )
        })}</div>
      </div>
    )
  }
}

function mapStateToProps (state) {
  return {
    feke: state.feke
  }
}

export default connect(mapStateToProps)(FekeStatus)
