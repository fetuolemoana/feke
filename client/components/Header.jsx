import React from 'react'
import {Link} from 'react-router-dom'

const Header = () => {
  return (
    <div>
      <Link to = {`/category`}>
        <h1>GO Feke!</h1>
      </Link>
    </div>
  )
}

export default Header
