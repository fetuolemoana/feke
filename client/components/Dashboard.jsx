import React from 'react'
import { Image, Container } from 'semantic-ui-react'
import FekeStatus from './FekeStatus'
// import { Link } from 'react-router-dom'
import { connect } from 'react-redux'


class Dashboard extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            name: 'Micah',
            village: 'Hakupu'
        }
    }




    render() {
        return (
            <div className='dashboard-page'>
                <FekeStatus />
                <Container textAlign='justified' centered="true" >
                    <div className='dashboard-container'>
                        {/* LINKS TO CATEGORY */}
                        <div className='dashboard-btn'>
                            {/* <Link to={`/greetings`}> */}
                            <button className="dashboard-btn ui purple button" >GREETINGS</button>

                            {/* </Link> */}
                        </div>
                        <div className='dashboard-btn'>
                            {/* <Link to={`/family`}> */}
                            <button className="dashboard-btn ui purple button" >FAMILY</button>

                            {/* </Link> */}
                        </div>

                        <div className='dashboard-btn'>
                            {/* <Link to={'/food'}> */}
                            <button className="dashboard-btn ui purple button" >FOOD</button>

                            {/* </Link> */}
                        </div>

                        <div className='dashboard-btn'>
                            {/* <Link to={'/numbers'}> */}
                            <button className="dashboard-btn ui purple button" >NUMBERS</button>
                            {/* </Link> */}
                        </div>
                    </div>

                </Container>
            </div >
        )
    }
}

// const mapDispatchToProps = (dispatch) => {
//     return {
//         //   placeHolderProps: (placeholderCategoryName) => {
//         // dispatch(placeHolderAction(placeholderCategoryName))
//         //   }
//     }
// }

// connect(mapDispatchToProps)()Dashboard
export default Dashboard