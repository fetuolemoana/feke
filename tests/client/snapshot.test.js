import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { shallow, render } from 'enzyme'
import Header from '../../client/components/Header'
import Dashboard from '../../client/components/Dashboard'
import CreateFeke from '../../client/components/CreateFeke'
import FekeStatus from '../../client/components/FekeStatus'


test('<Dashboard />', () => {
  const mockStore = configureStore()({ isUpdated: false })
  const wrap = shallow(
    <Provider store={mockStore}>
      <Dashboard />)
    </Provider>
  )
  const component = wrap.text()

  expect(renderer.create(component).toJSON()).toMatchSnapshot()
})

test('<CreateFeke />', () => {
  const mockStore = configureStore()({
    feke: [
      {
        id: 1,
        category_id: 1,
        name: 'Fekefeke',
        village: 'Avatele',
        image: 'feke-blue.png',
        status: 'Level 1'
      }
    ]
  })
  const wrap = shallow(
    <Provider store={mockStore}>
      <CreateFeke />)
    </Provider>
  )
  const component = wrap

  expect(renderer.create(component).toJSON()).toMatchSnapshot()
})



test('<Header />', () => {
  const wrap = shallow(<Header />)
  const component = wrap.text()
  expect(renderer.create(component).toJSON()).toMatchSnapshot()
})



test('<CreateFeke />', () => {
  const mockStore = configureStore()({ isUpdated: false })
  const wrap = shallow(
    <Provider store={mockStore}>
      <CreateFeke />)
    </Provider>
  )
  const component = wrap

  expect(renderer.create(component).toJSON()).toMatchSnapshot()
})