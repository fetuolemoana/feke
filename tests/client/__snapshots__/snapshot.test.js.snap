// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateFeke /> 1`] = `
Array [
  <div
    className="createfeke-page"
  >
    <h3
      className="validateInput"
    >
      
    </h3>
    <h1>
      Mitaki! Select and name your feke
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        className="createfeke-container"
      >
        <img
          className="createfeke-image"
          name="feke-blue.png"
          onClick={[Function]}
          src="images/feke-blue.png"
        />
        <img
          className="createfeke-image"
          name="feke-green.png"
          onClick={[Function]}
          src="images/feke-green.png"
        />
        <img
          className="createfeke-image"
          name="feke-purple.png"
          onClick={[Function]}
          src="images/feke-purple.png"
        />
        <img
          className="createfeke-image"
          name="feke-red.png"
          onClick={[Function]}
          src="images/feke-red.png"
        />
      </div>
      <div
        className="createfeke-container"
      >
        <label
          htmlFor="name"
        >
          Name 
        </label>
        <input
          name="name"
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          htmlFor="village"
        >
          Village 
        </label>
        <select
          className="createfeke-select"
          onChange={[Function]}
          value="Avatele"
        >
          <option
            village="alofi"
          >
            Alofi
          </option>
          <option
            village="avatele"
          >
            Avatele
          </option>
          <option
            village="hakupu"
          >
            Hakupu
          </option>
          <option
            village="hikutavake"
          >
            Hikutavake
          </option>
          <option
            village="lakepa"
          >
            Lakepa
          </option>
          <option
            village="liku"
          >
            Liku
          </option>
          <option
            village="makefu"
          >
            Makefu
          </option>
          <option
            village="mutalau"
          >
            Mutalau
          </option>
          <option
            village="namukulu"
          >
            Namukulu
          </option>
          <option
            village="tamakautoga"
          >
            Tamakautoga
          </option>
          <option
            village="toi"
          >
            Toi
          </option>
          <option
            village="tuapa"
          >
            Tuapa
          </option>
          <option
            village="vaiea"
          >
            Vaiea
          </option>
        </select>
        <input
          className="createfeke-submit ui purple button"
          type="submit"
          value="GO! FEKE"
        />
      </div>
    </form>
  </div>,
  ")",
]
`;

exports[`<CreateFeke /> 2`] = `
Array [
  <div
    className="createfeke-page"
  >
    <h3
      className="validateInput"
    >
      
    </h3>
    <h1>
      Mitaki! Select and name your feke
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        className="createfeke-container"
      >
        <img
          className="createfeke-image"
          name="feke-blue.png"
          onClick={[Function]}
          src="images/feke-blue.png"
        />
        <img
          className="createfeke-image"
          name="feke-green.png"
          onClick={[Function]}
          src="images/feke-green.png"
        />
        <img
          className="createfeke-image"
          name="feke-purple.png"
          onClick={[Function]}
          src="images/feke-purple.png"
        />
        <img
          className="createfeke-image"
          name="feke-red.png"
          onClick={[Function]}
          src="images/feke-red.png"
        />
      </div>
      <div
        className="createfeke-container"
      >
        <label
          htmlFor="name"
        >
          Name 
        </label>
        <input
          name="name"
          onChange={[Function]}
          type="text"
          value=""
        />
        <label
          htmlFor="village"
        >
          Village 
        </label>
        <select
          className="createfeke-select"
          onChange={[Function]}
          value="Avatele"
        >
          <option
            village="alofi"
          >
            Alofi
          </option>
          <option
            village="avatele"
          >
            Avatele
          </option>
          <option
            village="hakupu"
          >
            Hakupu
          </option>
          <option
            village="hikutavake"
          >
            Hikutavake
          </option>
          <option
            village="lakepa"
          >
            Lakepa
          </option>
          <option
            village="liku"
          >
            Liku
          </option>
          <option
            village="makefu"
          >
            Makefu
          </option>
          <option
            village="mutalau"
          >
            Mutalau
          </option>
          <option
            village="namukulu"
          >
            Namukulu
          </option>
          <option
            village="tamakautoga"
          >
            Tamakautoga
          </option>
          <option
            village="toi"
          >
            Toi
          </option>
          <option
            village="tuapa"
          >
            Tuapa
          </option>
          <option
            village="vaiea"
          >
            Vaiea
          </option>
        </select>
        <input
          className="createfeke-submit ui purple button"
          type="submit"
          value="GO! FEKE"
        />
      </div>
    </form>
  </div>,
  ")",
]
`;

exports[`<Dashboard /> 1`] = `"<Dashboard />)"`;

exports[`<Header /> 1`] = `"<Link /><Link />"`;
